lambdas/initialise-ipv-session/src/test/java/uk/gov/di/ipv/core/initialiseipvsession/InitialiseIpvSessionHandlerTest.java

```diff
@@ -345,6 +356,59 @@ void shouldReturnIpvSessionIdWhenRecoverableErrorFound()
         assertEquals(ipvSessionItem.getIpvSessionId(), responseBody.get("ipvSessionId"));
     }
 
+    @Test
+    void shouldValidateAndStoreAnyInheritedIdentity()
+            throws JsonProcessingException, JarValidationException, ParseException,
+                    VerifiableCredentialException, InvalidKeySpecException,
+                    NoSuchAlgorithmException, JOSEException {
+        // Arrange
+        when(mockIpvSessionService.generateIpvSession(any(), any(), any()))
+                .thenReturn(ipvSessionItem);
+        when(mockClientOAuthSessionDetailsService.generateClientSessionDetails(any(), any(), any()))
+                .thenReturn(clientOAuthSessionItem);
+        JWTClaimsSet claimsSet =
+                new JWTClaimsSet.Builder()
+                        .expirationTime(new Date(Instant.now().plusSeconds(1000).getEpochSecond()))
+                        .issueTime(new Date())
+                        .notBeforeTime(new Date())
+                        .subject("test-user-id")
+                        .audience("test-audience")
+                        .issuer("test-issuer")
+                        .claim("response_type", "code")
+                        .claim("redirect_uri", "http://example.com")
+                        .claim("state", "test-state")
+                        .claim("client_id", "test-client")
+                        .claim("vtr", List.of("Cl.Cm.P2", "Cl.Cm.PCL200"))
+                        .claim(
+                                "https://vocab.account.gov.uk/v1/inheritedIdentityJWT",
+                                getInheritedIdentityJWT().serialize())
+                        .build();
+        when(mockJarValidator.validateRequestJwt(any(), any())).thenReturn(claimsSet);
+        when(mockConfigService.enabled(CoreFeatureFlag.INHERITED_IDENTITY))
+                .thenReturn(true); // Mock enabled inherited identity feature flag
+        CriConfig testCriConfig =
+                CriConfig.builder()
+                        .componentId("test-component-id")
+                        .signingKey("test-signing-key")
+                        .build();
+        when(mockConfigService.getCriConfig(HMRC_MIGRATION_CRI)).thenReturn(testCriConfig);
+
+        APIGatewayProxyRequestEvent event = new APIGatewayProxyRequestEvent();
+        Map<String, Object> sessionParams =
+                Map.of("clientId", "test-client", "request", signedEncryptedJwt.serialize());
+        event.setBody(objectMapper.writeValueAsString(sessionParams));
+        event.setHeaders(Map.of("ip-address", TEST_IP_ADDRESS));
+
+        // Act
+        initialiseIpvSessionHandler.handleRequest(event, mockContext);
+
+        // Assert
+        verify(mockVerifiableCredentialJwtValidator, times(1))
+                .validate(any(), eq(testCriConfig), eq("test-user-id"));
```

Comment:
Ah if this method were actually called it would fail because the inherited identity sub !== "test-user-id"